plugins/woocommerce-admin/client/core-profiler/index.tsx

```diff
@@ -681,7 +682,12 @@ export const CoreProfilerController = ( {
 	const CurrentComponent =
 		currentNodeMeta?.component ??
 		( () => (
-			<div data-testid="core-profiler-loading-screen">Insert Spinner</div>
+			<div
```

Comment:
I had intended to move this out to its own file when thinking about this issue, as you can see the existing syntax is a bit ugly.

This is because React component definitions are functions that return JSX, hence the need to wrap the JSX here with () => ( JSX ).

If it's moved out to a named function and/or another file, then this would just look like 
```
const CurrentComponent = currentNodeMeta?.component ?? (<ProfilerSpinner />)
```

Perhaps a good place for this would be in the `./core-profiler/components` folder along with the Choice and Heading components?

Reply:
Thank you for the suggestion here; I just updated the branch accordingly.